src/java/com/sarality/app/view/ViewDataLoaderTask.java

```diff
@@ -0,0 +1,27 @@
+package com.sarality.app.view;
+
+import android.content.Context;
+import android.support.v4.content.AsyncTaskLoader;
+
+import com.sarality.app.view.datasource.DataSource;
+
+/**
+ * Asynchronous loader that loads the data asynchronously.
+ *
+ * @author abhideep@ (Abhideep Singh)
+ */
+public class ViewDataLoaderTask<T> extends AsyncTaskLoader<T> {
+
+  private final DataSource<T> dataSource;
+
+  public ViewDataLoaderTask(Context context, DataSource<T> dataSource) {
+    super(context);
+    this.dataSource = dataSource;
+  }
+
+  @Override
+  public T loadInBackground() {
+    dataSource.loadData();
```

Comment:
How will we customise this .. for example in case when the data is to be picked from two different tables